inc/migration.class.php

```diff
@@ -341,10 +343,14 @@ function addField($table, $field, $type, $options=array()) {
          $params['first'] = " FIRST ";
       }
 
+      if ($params['null']) {
+         $params['null'] = 'NULL ';
+      }
+
       if ($format) {
          if (!FieldExists($table, $field, false)) {
             $this->change[$table][] = "ADD `$field` $format ".$params['comment'] ." ".
-                                           $params['first'].$params['after']."";
+                                      $params['first'].$params['after'];
```

Comment:
after your last changes, you miss to redo the insertion of your NULL param in the query :)